### PR TITLE
Avoid call to get cloud instances

### PR DESCRIPTION
**Pull request by dims**
**What this PR does / why we need it**:

if a node does not have the taint, we really don't need to make calls
to get the list of instances from the cloud provider

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
Found when reviewing code for #59887

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/controller/cloud/node_controller.go [hunk 1]
```diff
@@ -313,18 +313,18 @@ func (cnc *CloudNodeController) MonitorNode() {
...
-		return
-	}
-
 	cloudTaint := getCloudTaint(node.Spec.Taints)
 	if cloudTaint == nil {
 		glog.V(2).Infof("This node %s is registered without the cloud taint. Will not process.", node.Name)
 		return
 	}
 
+	instances, ok := cnc.cloud.Instances()
```
Comment (wlan0) [inline pkg/controller/cloud/node_controller.go]: :+1:

**Comment by jhorwit2** [comment]
@dims @wlan0 I don't think this does what you expect. `cnc.cloud.Instances()` doesn't actually call the cloud; it just returns the Instances interface implementation for the cloud and true/false whether it's supported for the given cloud.

**Comment by dims** [comment]
@jhorwit2 ack, should we leave it alone then? or should i revert?

**Pull request merged**